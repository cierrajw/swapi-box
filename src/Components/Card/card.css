.card{
  display: inline-block;
  margin: 1rem;
  background-color: yellow;
  width: 13rem;
  height: 20rem;
  /* clip-path: polygon(50% 0, 66% 28%, 98% 35%, 79% 59%, 93% 100%, 49% 89%, 6% 100%, 20% 61%, 2% 35%, 33% 29%); */
}
