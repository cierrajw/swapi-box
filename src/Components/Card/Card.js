import React from 'react';
import PropTypes from 'prop-types';
import './card.css';

const Card = ({ 
  name, 
  homeworld, 
  species, 
  language, 
  population, 
  model, 
  vehicleClass, 
  passengers, 
  terrain, 
  planetPopulation, 
  climate, 
  residents, 
  id, 
  favorite, 
  type, 
  toggleFavorite }) => {

  switch (type){

    case 'people':
      return (
        <div 
          className={favorite ? "card-favorite": "card"}
          onClick={()=>toggleFavorite(id, type)}
        >
          <section
            className="favorite-name"><h2>{name}</h2>
            <div
              className={favorite ? "star-favorite": "star"}
            >
            </div>
          </section>

          <h4>Homeworld: {homeworld}</h4>
          <h4>Species: {species}</h4>
          <h4>Language: {language}</h4>
          <h4>Population: {population}</h4>
        </div>
      );

    case 'vehicles':
      return (
        <div 
          className={favorite ? "card-favorite": "card"}
          onClick={()=>toggleFavorite(id, type)}
        >
          <section
            className="favorite-name"><h2>{name}</h2>
            <div
              className={favorite ? "star-favorite": "star"}
            >.
            </div>
          </section>
          <h4>Model: {model}</h4>
          <h4>Class: {vehicleClass}</h4>
          <h4># of Passengers: {passengers}</h4>
        </div>
      );

    case 'planets':

      let residentResult = residents.map(resident => {
        if (residents.length > 1) {
          return <h4 className="resident-list"> {resident}, </h4>;
        } else {
          return <h4> {resident} </h4>;
        } 
      });
      
      return (
        <div 
          className={favorite ? "card-favorite": "card"}
          onClick={()=>toggleFavorite(id, type)}
        >
          <section
            className="favorite-name"><h2>{name}</h2>
            <div
              className={favorite ? "star-favorite": "star"}
            >.
            </div>
          </section>
          <h4>Terrain: {terrain}</h4>
          <h4>Population: {planetPopulation}</h4>
          <h4>Climate: {climate}</h4>
          <h4 className="residents">Residents:
            {residentResult}
          </h4>
        </div>
      );

    default:
      break;
  } 
};


Card.propTypes = {
  name: PropTypes.string,
  homeworld: PropTypes.string,
  species: PropTypes.string,
  language: PropTypes.string,
  population: PropTypes.string,
  model: PropTypes.string,
  vehicleClass: PropTypes.string,
  passengers: PropTypes.string,
  terrain: PropTypes.string,
  planetPopulation: PropTypes.string,
  climate: PropTypes.string,
  residents: PropTypes.array,
  id: PropTypes.number,
  type: PropTypes.string,
  toggleFavorite: PropTypes.func,
  favorite: PropTypes.bool
};

export default Card;
