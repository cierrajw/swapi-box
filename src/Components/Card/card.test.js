import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import Card from './Card.js';

describe('Card', () => {

  it('should match the snapshot', () => {

    const wrapper = shallow(<Card />);

    expect(wrapper).toMatchSnapshot();

  });

});
