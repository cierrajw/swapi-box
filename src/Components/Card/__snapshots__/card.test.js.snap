// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <h4>
          Homeworld: 
        </h4>,
        <h4>
          Species: 
        </h4>,
        <h4>
          Language: 
        </h4>,
        <h4>
          Population: 
        </h4>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Homeworld: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Homeworld: ",
          undefined,
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Species: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Species: ",
          undefined,
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Language: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Language: ",
          undefined,
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Population: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Population: ",
          undefined,
        ],
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <h4>
            Homeworld: 
          </h4>,
          <h4>
            Species: 
          </h4>,
          <h4>
            Language: 
          </h4>,
          <h4>
            Population: 
          </h4>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Homeworld: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Homeworld: ",
            undefined,
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Species: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Species: ",
            undefined,
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Language: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Language: ",
            undefined,
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Population: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Population: ",
            undefined,
          ],
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
